File: Leetcode_solutions/search-insert-position.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
  //   loop in, if there is target in nums, return the index
  // else
  //     find the bigger value than target
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] === target) {
      return i;
    }
  }
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] > target) {
      return i;
    } else if (i === nums.length - 1) {
      return i + 1;
    }
  }
};

File: Leetcode_solutions/search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        pos = None
        if len(nums) == 0:
            return 0
        elif len(nums) == 1:
            if nums[0] == target:
                return 0
            elif nums[0] < target:
                return 1
            else:
                return 0
        else:
            for i in range(1, len(nums)):
                if nums[i] == target:
                    pos = i
                    break
                elif nums[i - 1] < target and nums[i] > target:
                    pos = i
                    break
                elif i == len(nums) - 1 and target > nums[i]:
                    pos = i + 1
                else:
                    pos = 0
            return pos
